serialization/dotnet/json/src/JsonSerializationWriter.cs

```diff
@@ -16,7 +16,7 @@ namespace Microsoft.Kiota.Serialization.Json
     /// <summary>
     /// The <see cref="ISerializationWriter"/> implementation for json content types.
     /// </summary>
-    public class JsonSerializationWriter : ISerializationWriter, IDisposable
+    public class JsonSerializationWriter : ISerializationWriter
```

Comment:
minor cleanup: `ISerializationWriter` already implements `IDisposable`

Reply:
I have somewhere in the corner of my mind that IDisposable should not be "hidden" to make it easier to people and tooling to discover, but this is probably old information as [the only recommendation for inheritance is this](https://docs.microsoft.com/en-us/dotnet/api/system.idisposable?view=net-5.0#idisposable-and-the-inheritance-hierarchy). :)

Reply:
I hadn't thought about it like that. I see the value in keeping it as is and have reverted it via https://github.com/microsoft/kiota/pull/529/commits/fd02ffd086434992b122cf12ae0aa9b441741815.